File: CartoMap.Objective-C/CartoMap.Objective-C/MainViewController.m
//
//  MainViewController.m
//  CartoMap.Objective-C
//
//  Created by Aare Undo on 28/09/2017.
//  Copyright © 2017 Aare Undo. All rights reserved.
//

#import "MainViewController.h"

@interface MainViewController ()
@end

@implementation MainViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.navigationItem.title = @"CARTO MAP";
    
    self.contentView = [[MainView alloc] init];
    self.view = self.contentView;
    
    NSMutableArray *list = [Samples getList];
    [self.contentView addRowsWithRows:list];
}

- (void) viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    
    self.contentView.galleryDelegate = self;
}

- (void) viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];
    
    self.contentView.galleryDelegate = nil;
}

- (void) galleryItemClickWithItem:(GalleryRow *)item {
    
    // Launch selected sample, use basic reflection to convert class name to class instance
    UIViewController* controller = [[NSClassFromString(item.sample.controller) alloc] init];
    [controller setTitle: item.sample.title];
    [[self navigationController] pushViewController:controller animated:true];
}

@end

File: AdvancedMap.Swift/Feature Demo/MainViewController.swift
//
//  ViewController.swift
//  Feature Demo
//
//  Created by Aare Undo on 16/06/2017.
//  Copyright © 2017 CARTO. All rights reserved.
//

import UIKit

class MainViewController: UIViewController, GalleryDelegate {
    
    var contentView: MainView?
    
    override func viewDidLoad() {
        super.viewDidLoad()
    
        contentView = MainView()
        contentView?.galleryDelegate = self
        
        view = contentView

        contentView?.addRows(rows: Samples.list)
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        title = "FEATURE DEMO"
    }
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
        title = ""
    }
    
    func galleryItemClick(item: GalleryRow) {
        
        let controller = item.sample.controller!
        controller.sample = item.sample
        controller.title = item.sample.title
        
        navigationController?.pushViewController(controller, animated: true)
    }

}

